### ISSUE TITLE
Exclusions Applications

### ISSUE THREAD

**Issue by delphinuz** [MSF-Jarvis/wireguard-kt#11]
Is that possible including all applications(include freezed) in the list?

Thanks.

**Comment by MSF-Jarvis** [comment]
Looking at [the code](https://github.com/MSF-Jarvis/wireguard-kt/blob/d11db7a1da3fb85549cfae732c1ee7d96450a9d4/app/src/main/java/com/wireguard/android/fragment/AppListDialogFragment.kt#L70-L72), only apps with a launcher activity are indexed there. Are you __absolutely__ sure you want more than that? Can you list a few examples if it's not a hassle?

**Comment by delphinuz** [comment]
>  Maybe some kind of opt-in behavior via settings?
That will be a good idea if possible.

I always freezed some apps(for example: Facebook, Taobao, QQ, Alipay, Google Play,...)that taking lots of resources in the background. Those apps don't show up in the Exclusions Applications list until they are actived.

It bother me everytime when I add a new wireguard server or install/remove an app that shoud be in or already in the Exclusions Applications list. I have to active echo of apps that freezed and in the list, switch to wireguard gui, add them (include new one) in the list, save,  switch back to freeze app and freeze echo of them. 

That's why I really need this function be implemented. Thanks.

**Comment by MSF-Jarvis** [comment]
Okay so, frozen apps are the current pet peeve here and not system apps? Can you update the issue title to reflect that you want frozen apps to show in the list?

**Comment by delphinuz** [comment]
> Okay so, frozen apps are the current pet peeve here and not system apps? 
Most of apps are user apps, some are system apps such as Google Play Store.

**Comment by MSF-Jarvis** [comment]
> Most of apps are user apps, some are system apps such as Google Play Store.

Currently the query filters all apps which can be launched through your launcher, is that enough for your use-case?

**Comment by MSF-Jarvis** [comment]
> Just want freezen apps show up in the EA list(global or individual). They don't show up in last buld(0.0.20180826).

Yes I know, that's why this issue is still open. Also, binaries built off this fork are __not__ distributed publicly yet, the Play Store listing [here](https://play.google.com/store/apps/details?id=com.wireguard.android) is of the upstream project which is hosted [here](https://git.zx2c4.com/wireguard-android/). Just an FYI in case you're mixing things up.

**Comment by MSF-Jarvis** [comment]
![image](https://user-images.githubusercontent.com/13348378/44996607-694f7300-afc6-11e8-9f3d-19e2eb5fb6b4.png)

Things like this will happen if I allow straight up allow disabled components in the list. That's all activities of the app that stay disabled until you switch the app icon color (as evident). Still trying to see if there's another way around but that's the current status.

**Comment by delphinuz** [comment]
FYI https://github.com/Playhi/FreezeYou/blob/25d273eee7dbd06578c10c805dcd08cee5326746/app/src/main/java/cf/playhi/freezeyou/Main.java#L343

**Comment by MSF-Jarvis** [comment]
Awesome.
